lib/screens/v2/widget_instance/elevator_closures.ex

```diff
@@ -32,15 +25,111 @@ defmodule Screens.V2.WidgetInstance.ElevatorClosures do
           }
   end
 
+  defmodule Upcoming do
+    @moduledoc false
+
+    @enforce_keys ~w[period]a
+    defstruct @enforce_keys ++ ~w[summary]a
+
+    @type t :: %__MODULE__{period: {Date.t(), Date.t() | nil}, summary: String.t() | nil}
+
+    @nbsp "\u00A0"
+
+    @spec serialize(t(), DateTime.t()) :: map()
+    def serialize(%__MODULE__{period: {start_date, end_date}, summary: summary}, now) do
+      today = Util.service_date(now)
+      indefinite? = is_nil(end_date)
+      starts_tomorrow? = Date.diff(start_date, today) == 1
+      starts_this_week? = Date.beginning_of_week(start_date) == Date.beginning_of_week(today)
+      relative_start? = starts_tomorrow? or starts_this_week?
+      single_day? = start_date == end_date
+
+      same_month? =
+        not is_nil(end_date) and
+          {start_date.year, start_date.month} == {end_date.year, end_date.month}
```

Comment:
Could be simplified b/c you already set `indefinite? = is_nil(end_date)` 
So this could be:
```
same_month? = not indefinite && {start_date.year, start_date.month} == {end_date.year, end_date.month}
```

Reply:
I think I did it this way to try and clarify that the reason for the short-circuit has nothing to do with however `indefinite?` is defined, but is an explicit `nil` guard for the `end_date.x` field access that comes after it in the conditional. This might change anyway based on [your suggestion here](https://github.com/mbta/screens/pull/2443#discussion_r1968115187) though.